[Chen Shaoheng] iP

cshao02 opened the pull request:
# Gojou - Your Personal Task Manager 🤖

> "The key is not to prioritize what's on your schedule, but to schedule your priorities." ~Stephen Covey [(source)](https://www.linkedin.com/pulse/key-prioritize-whats-your-schedule-priorities-stephen-emma-g--tus9c/)

**Gojou** is a friendly chatbot that helps you manage your tasks efficiently. You can store to-do items, set deadlines, and keep track of events. Gojou also lets you query, list, and delete tasks with simple commands. It's

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

And it's **FREE**! 

## Features
- [x] Store different types of tasks: to-dos, deadlines, and events.
- [X] Query tasks to see what's on your plate.
- [x] Get a list of all your tasks in one place.
- [X] Mark tasks as done or unmark them.
- [X] Delete tasks you no longer need.

## How to Use Gojou

1. **Add a Task:** You can add a task by telling Gojou the task type, description and dates (if applicable)
2. **View Tasks:** Ask Gojou to show all tasks, and you’ll get a list of everything you’ve added.
3. **Query Tasks:** Use specific keywords to find particular tasks.
4. **Delete a Task:** Tell Gojou which task you want to remove.
5. **Mark/Unmark a Task:** Get Gojou to mark tasks that are done or unmark them if they are not.
##
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        new Gojou("./data/Gojou.txt").run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

cshao02 committed:
Renamed chatbot and changed greeting

cshao02 committed:
Added echo feature

cshao02 committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

cshao02 committed:
Add the ability to mark tasks as done or undone

cshao02 committed:
Added Todo class

cshao02 committed:
Added Deadline and Event classes

cshao02 committed:
Before changing to scanner.next()

cshao02 committed:
Made Todo and Deadline work

cshao02 committed:
Finished todo, deadline and event

cshao02 committed:
Added testing

cshao02 committed:
Created Exception

cshao02 committed:
Handled exceptions for todo and deadline

cshao02 committed:
Handled most errors

cshao02 committed:
Added delete function

cshao02 committed:
Add write to file capabilities

cshao02 committed:
Complete the save feature

cshao02 committed:
Merge branch 'branch-Level-7'

cshao02 committed:
Teach chatbot to understand date and time for deadline tasks

cshao02 committed:
Teach chatbot to understand date and time for event tasks

cshao02 committed:
Add LocalDateTime to Deadline and Event classes

cshao02 committed:
Merge branch 'branch-Level-8'

cshao02 committed:
Use more OOP

The code is cluttered in one file called Alex.java.

This makes the code hard to read and hard to maintain.

Let's refactor the code to extract out closely related code as classes
to enhance code readability.

cshao02 committed:
Merge branch 'branch-A-MoreOOP'

cshao02 committed:
Put all files under package alex

cshao02 committed:
Create task package

cshao02 committed:
Add command package

cshao02 committed:
Merge branch 'branch-A-Packages'

cshao02 committed:
Merge branch 'add-gradle-support'

cshao02 committed:
Add Gradle

cshao02 committed:
Add test case for Todo task

cshao02 committed:
Add test cases for Event and Deadline tasks

cshao02 committed:
Merge branch 'branch-A-JUnit'

cshao02 committed:
Create JAR file

cshao02 committed:
Add JavaDoc comments

cshao02 committed:
Improve code to adhere to coding standard

cshao02 committed:
Add search functionality

cshao02 committed:
Merge branch 'branch-A-JavaDoc'

cshao02 committed:
Fix merge conflicts

cshao02 committed:
Fix merge conflicts

cshao02 committed:
Update Alex.txt

cshao02 committed:
Add JavaFX GUI

tanwartushar commented on src/main/java/alex/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+package alex;
```
Since `TaskList` is somewhat related to the `task` package, could it perhaps be added to it?

cshao02 replied:
Good point! Thanks for that, I'll incorporate that change

tanwartushar commented on src/main/java/alex/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+        System.out.println(line);
+    }
+
+    public void findWord(String string, Ui ui) {
```
You could add JavaDocs here :)

tanwartushar commented on src/main/java/alex/task/Event.java:
```diff
@@ -0,0 +1,36 @@
+public class Event extends Task {
+    private LocalDateTime start;
+
+    private LocalDateTime end;
```
Although the variable names `start` and `end` are understandable, you could perhaps use more apt names like `startTime` and `endTime`.

cshao02 replied:
Thanks for the feedback, will make my variable names more descriptive

tanwartushar commented on src/main/java/alex/command/Command.java:
```diff
@@ -0,0 +1,32 @@
+package alex.command;
```
Nice work on extracting all the `command` classes!

tanwartushar reviewed (commented):
Looks great!

cshao02 committed:
Tweak GUI for better design

cshao02 committed:
Merge branch 'branch-Level-10'

cshao02 committed:
Add CheckStyle

cshao02 committed:
Change the code to follow issues raised by CheckStyle

cshao02 committed:
Merge branch 'branch-A-CheckStyle'

cshao02 committed:
Clean up code

cshao02 committed:
Change data filepath

cshao02 committed:
Add assertions

The code has no assertions to check important assumptions
at various points of the program execution.

This causes possible runtime errors due to programmer
mistakes.

Let's add some assertions to document important assumptions
that should hold at various points in the code for additional
safety.

cshao02 committed:
Improve code quality

The code suffers from complicated expressions and poor ordering of
statements as well as a lack of code comments.

These make it hard to read and understand what the code is doing
at certain points.

Let's simplify some code expressions, put the guard clause in front
to make the happy path prominent and add some comments to explain
what the code is doing to enhance overall readability of the code.

cshao02 committed:
Merge pull request #1 from cshao02/branch-A-Assertions

Add assertions

cshao02 committed:
Merge branch 'master' into branch-A-CodeQuality

cshao02 committed:
Merge pull request #2 from cshao02/branch-A-CodeQuality

Improve code quality

cshao02 committed:
Fix bug for makeTodoTask

cshao02 committed:
Set up CI

cshao02 committed:
Add priority to tasks

cshao02 committed:
Merge branch 'branch-BCD-Extension'

cshao02 committed:
Remove CI file

cshao02 committed:
Update test cases

cshao02 committed:
Add CI file again

cshao02 committed:
Fix file location in JAR file

cshao02 committed:
Fix handling of file path for JAR file

cshao02 committed:
Add icon

cshao02 committed:
Rename Chatbot and change it's associated personality

cshao02 committed:
Update README

cshao02 committed:
Merge branch 'branch-A-Personality'

cshao02 committed:
Edit README

cshao02 committed:
Update README.md

cshao02 committed:
Refactor code for better readability

cshao02 committed:
Fix style errors

cshao02 committed:
Merge branch 'master' of https://github.com/cshao02/ip

cshao02 committed:
Change event creation logic
